File: leetcode_python/easy/ReverseInteger.py
'''
Created on Dec 21, 2014

@author: Chandra Kant
'''
class Solution:
    # @return an integer
    def reverse(self, x):
        tmp = abs(x);
        num = 0;
        max_value = ((1 << 31) - 1);
        while tmp != 0:
            md = tmp % 10;
            num = num * 10 + md;
            tmp = int(tmp / 10);
        if num > max_value:
            num = 0;
        if x < 0:
            num = num * -1;
        return int(num);     
if __name__ == '__main__':
    sol = Solution();
    print(sol.reverse(1 << 31 - 1));
    print(sol.reverse(0));
    print(sol.reverse(100));
    print(sol.reverse(1234));
    print(sol.reverse(-1234));
    print(sol.reverse(156));
    print(sol.reverse(-2147483412));

File: leetcode/src/easy/ReverseInteger.java
package easy;
public class ReverseInteger {
	public int reverse(int x) {
		int tmp = Math.abs(x);
		long num = 0;
		while (tmp != 0) {
			int mod = tmp % 10;
			num = num * 10 + mod;
			tmp /= 10;
		}
		if (num > Integer.MAX_VALUE || num < Integer.MIN_VALUE)
			num = 0;
		else if (x < 0)
			num *= -1;	
		return (int)num;
	}

	public static void main(String[] args) {
		ReverseInteger sol = new ReverseInteger();
		System.out.println(sol.reverse(1324));
		System.out.println(sol.reverse(0));
		System.out.println(sol.reverse(100));
		System.out.println(sol.reverse(Integer.MAX_VALUE));
		System.out.println(sol.reverse(Integer.MIN_VALUE));
		System.out.println(sol.reverse(-2147483412));
	}
}
